testing/test_mercurial.py

```diff
@@ -108,3 +108,48 @@ def test_version_in_merge(wd):
 def test_parse_no_worktree(tmpdir):
     ret = parse(str(tmpdir))
     assert ret is None
+
+
+@pytest.fixture
+def version_1_0(wd):
+    wd('hg branch default')
+    wd.commit_testfile()
+    wd('hg tag 1.0 -u test -d "0 0"')
+    return wd
+
+
+@pytest.fixture
+def pre_merge_commit_after_tag(version_1_0):
+    wd = version_1_0
+    wd('hg branch testbranch')
+    wd.write('branchfile', 'branchtext')
+    wd(wd.add_command)
+    wd.commit()
+    wd('hg update default')
+    wd('hg merge testbranch')
+    return wd
+
+
+def test_version_bump_before_merge_commit(pre_merge_commit_after_tag):
```

Comment:
the fixture naming interaction seems a bit unnatural to me since those fixtures are actions that build up the content of a different fixture

i would propose to always add the wd fixture to the argument set, and to use the use-fixtures marker in tests

Reply:
Oh, I hadn't seen the usefixtures marker before - will change it, that's much clearer 👍